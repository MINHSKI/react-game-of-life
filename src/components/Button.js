import styled from 'styled-components';

const Button = styled.button`
  background-color: black;
  color: white;
  border: none;
  border-radius: 0.3rem;
  font-weight: bold;
  display: inline;
  margin: 0.5rem 0.2rem;
  padding: 0.3rem 0.6rem;
`;

export default Button;
