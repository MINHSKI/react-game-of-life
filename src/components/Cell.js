import styled from 'styled-components';

const Cell = styled.div`
  background: ${props => (props.alive ? 'black' : 'white')};
  color: ${props => (props.alive ? 'white' : 'black')};
`;

export default Cell;
