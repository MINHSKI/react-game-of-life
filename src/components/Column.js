import styled from 'styled-components';

const Column = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding: 1rem;
`;

export default Column;
