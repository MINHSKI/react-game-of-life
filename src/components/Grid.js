import styled from 'styled-components';

const Grid = styled.div`
  display: grid;
  grid-gap: 1px;
  grid-template-rows: repeat(${props => props.rows || 25}, ${props => props.size || '8px'});
  grid-template-columns: repeat(${props => props.cols || 25}, ${props => props.size || '8px'});
  margin: 0 auto;
`;

export default Grid;
