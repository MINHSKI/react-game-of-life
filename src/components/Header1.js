import styled from 'styled-components';

const Header1 = styled.h1`
  margin: 0;
  font-size: 2rem;
  font-weight: bold;
`;

export default Header1;
