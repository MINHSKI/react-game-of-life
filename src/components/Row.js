import styled from 'styled-components';

const Row = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: left;
`;

export default Row;
