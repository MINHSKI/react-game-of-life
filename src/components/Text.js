import styled from 'styled-components';

const Text = styled.p`
  margin: 0 0 .7rem 0;
  font-size: 0.9rem;
`;

export default Text;
