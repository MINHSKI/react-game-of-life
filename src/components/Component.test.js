import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Button from './Button';

describe('Button', () => {
  test('should have the proper styles', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.type).toBe('button');
    expect(tree).toHaveStyleRule('border', 'none');
    expect(tree).toHaveStyleRule('font-weight', 'bold');
  });
});

import Cell from './Cell';

describe('Cell', () => {
  test('should have the proper styles when alive', () => {
    const tree = renderer.create(<Cell alive />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.type).toBe('div');
    expect(tree).toHaveStyleRule('background', 'black');
    expect(tree).toHaveStyleRule('color', 'white');
  });
  test('should have the proper styles when dead', () => {
    const tree = renderer.create(<Cell />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('background', 'white');
    expect(tree).toHaveStyleRule('color', 'black');
  });
});

import Column from './Column';

describe('Column', () => {
  test('should have the proper styles', () => {
    const tree = renderer.create(<Column />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.type).toBe('div');
    expect(tree).toHaveStyleRule('display', 'flex');
    expect(tree).toHaveStyleRule('flex-direction', 'column');
    expect(tree).toHaveStyleRule('justify-content', 'flex-start');
  });
});

import Grid from './Grid';

describe('Grid', () => {
  test('should have the proper styles', () => {
    const tree = renderer.create(<Grid />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.type).toBe('div');
    expect(tree).toHaveStyleRule('display', 'grid');
    expect(tree).toHaveStyleRule('grid-template-rows', 'repeat(25,8px)');
    expect(tree).toHaveStyleRule('grid-template-columns', 'repeat(25,8px)');
  });

  test('should have the right number of rows, columns and size', () => {
    const tree = renderer.create(<Grid rows={10} cols={99} size="20px" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.type).toBe('div');
    expect(tree).toHaveStyleRule('grid-template-rows', 'repeat(10,20px)');
    expect(tree).toHaveStyleRule('grid-template-columns', 'repeat(99,20px)');
  });
});

import Header1 from './Header1';

describe('Header1', () => {
  test('should have the proper styles', () => {
    const tree = renderer.create(<Header1 />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.type).toBe('h1');
    expect(tree).toHaveStyleRule('font-weight', 'bold');
  });
});

import Row from './Row';

describe('Row', () => {
  test('should have the proper styles', () => {
    const tree = renderer.create(<Row />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.type).toBe('div');
    expect(tree).toHaveStyleRule('display', 'flex');
    expect(tree).toHaveStyleRule('flex-direction', 'row');
    expect(tree).toHaveStyleRule('justify-content', 'left');
  });
});

import Text from './Text';

describe('Text', () => {
  test('should have the proper styles', () => {
    const tree = renderer.create(<Text />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.type).toBe('p');
  });
});
