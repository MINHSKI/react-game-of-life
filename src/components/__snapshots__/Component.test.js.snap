// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have the proper styles 1`] = `
.c0 {
  background-color: black;
  color: white;
  border: none;
  border-radius: 0.3rem;
  font-weight: bold;
  margin: 0.5rem 0.2rem;
  padding: 0.3rem 0.6rem;
}

<button
  className="c0"
/>
`;

exports[`Cell should have the proper styles when alive 1`] = `
.c0 {
  background: black;
  color: white;
}

<div
  className="c0"
/>
`;

exports[`Cell should have the proper styles when dead 1`] = `
.c0 {
  background: white;
  color: black;
}

<div
  className="c0"
/>
`;

exports[`Column should have the proper styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 1rem;
}

<div
  className="c0"
/>
`;

exports[`Grid should have the proper styles 1`] = `
.c0 {
  display: grid;
  grid-gap: 1px;
  grid-template-rows: repeat(25,8px);
  grid-template-columns: repeat(25,8px);
  margin: 0 auto;
}

<div
  className="c0"
/>
`;

exports[`Grid should have the right number of rows, columns and size 1`] = `
.c0 {
  display: grid;
  grid-gap: 1px;
  grid-template-rows: repeat(10,20px);
  grid-template-columns: repeat(99,20px);
  margin: 0 auto;
}

<div
  className="c0"
  cols={99}
  rows={10}
  size="20px"
/>
`;

exports[`Header1 should have the proper styles 1`] = `
.c0 {
  margin: 0;
  font-size: 2rem;
  font-weight: bold;
}

<h1
  className="c0"
/>
`;

exports[`Row should have the proper styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`Text should have the proper styles 1`] = `
.c0 {
  margin: 0 0 .7rem 0;
  font-size: 0.9rem;
}

<p
  className="c0"
/>
`;
